### DISCUSSION TITLE
How to draw a connection that is positioned underneath other objects?

### DISCUSSION THREAD

**Discussion by EllieAx** [jgraph/drawio-desktop#1546] [Q&A]
"I'm create diagrams using a Python script. First, the units are drawn, and then the connections between the units are added. The issue is that the connections are drawn over the units, making it difficult to interpret the diagram. Is there any way to make the connection underneath the objects when they are drawn? For example, a layering value in style."

**Comment by gammalogic** [comment]
Looking at the output of an XML file generated from a very basic chart, it seems that the order of items in the file is important e.g. all of the connections would have to appear at the start of the file so that they are drawn first, then the elements are drawn afterwards and will appear on top of the connections. Also, the elements must always be filled otherwise the connections will still be visible underneath. If you are doing this programmatically, perhaps you can reorder the physical position of the connections in your file?
